### DISCUSSION TITLE
Aggregation store usage

### DISCUSSION THREAD

**Discussion by thaingo** [yahoo/elide#2312] [Elide 5]
The Elide initialization does require a Jpa data store but I still want to double check if its possible to use only Aggregation store without the JPA data store.

**Comment by aklish** [comment]
You don’t need jpa store to use aggregation store.

**Reply by thaingo** [reply]
I excluded JPA store dependency and customized only one `DataStore` bean to use only aggregation store like this:
```
@Bean
    public DataStore buildDataStore(
        @Autowired(required = false) QueryEngine queryEngine,
        @Autowired(required = false) Cache cache,
        @Autowired(required = false) QueryLogger querylogger) {

        // aggregation store enabled
        AggregationDataStore.AggregationDataStoreBuilder aggregationDataStoreBuilder =
            AggregationDataStore.builder().queryEngine(queryEngine);
        // dynamic config enabled as well
        aggregationDataStoreBuilder.dynamicCompiledClasses(queryEngine.getMetaDataStore().getDynamicTypes());
        aggregationDataStoreBuilder.cache(cache);
        aggregationDataStoreBuilder.queryLogger(querylogger);
        AggregationDataStore aggregationDataStore = aggregationDataStoreBuilder.build();

        return new MultiplexManager(queryEngine.getMetaDataStore(), aggregationDataStore);
    }
```
But always face this exception `Caused by: java.lang.NoClassDefFoundError: com/yahoo/elide/datastores/jpa/transaction/JpaTransaction`

I just want to clarify if it is design intent. Its not a big deal to include JPA store dependency and customize/use the respective bean.

**Reply by aklish** [reply]
It is not the intent.  Probably is something wrong with Elide's pom in the spring boot starter.
